src/main/java/io/debezium/connector/jdbc/JdbcChangeEventSink.java

```diff
@@ -40,71 +42,181 @@ public class JdbcChangeEventSink implements ChangeEventSink {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(JdbcChangeEventSink.class);
 
+    public static final String SCHEMA_CHANGE_VALUE = "SchemaChangeValue";
+    public static final String DETECT_SCHEMA_CHANGE_RECORD_MSG = "Schema change records are not supported by JDBC connector. Adjust `topics` or `topics.regex` to exclude schema change topic.";
     private final JdbcSinkConnectorConfig config;
-    private final SessionFactory sessionFactory;
     private final DatabaseDialect dialect;
     private final StatelessSession session;
     private final TableNamingStrategy tableNamingStrategy;
+    private final RecordWriter recordWriter;
+
+    public JdbcChangeEventSink(JdbcSinkConnectorConfig config, StatelessSession session, DatabaseDialect dialect, RecordWriter recordWriter) {
 
-    public JdbcChangeEventSink(JdbcSinkConnectorConfig config) {
         this.config = config;
-        this.sessionFactory = config.getHibernateConfiguration().buildSessionFactory();
         this.tableNamingStrategy = config.getTableNamingStrategy();
-
-        this.dialect = DatabaseDialectResolver.resolve(config, sessionFactory);
-        this.session = this.sessionFactory.openStatelessSession();
-
+        this.dialect = dialect;
+        this.session = session;
+        this.recordWriter = recordWriter;
         final DatabaseVersion version = this.dialect.getVersion();
         LOGGER.info("Database version {}.{}.{}", version.getMajor(), version.getMinor(), version.getMicro());
     }
 
     @Override
-    public void execute(SinkRecord record) {
+    public void execute(Collection<SinkRecord> records) {
+
+        final Map<TableId, RecordBuffer> updateBufferByTable = new HashMap<>();
+        final Map<TableId, RecordBuffer> deleteBufferByTable = new HashMap<>();
+
+        for (SinkRecord record : records) {
+
+            LOGGER.trace("Processing {}", record);
+
+            validate(record);
+
+            Optional<TableId> optionalTableId = getTableId(record);
+            if (optionalTableId.isEmpty()) {
+
+                LOGGER.warn("Ignored to write record from topic '{}' partition '{}' offset '{}'. No resolvable table name", record.topic(), record.kafkaPartition(),
+                        record.kafkaOffset());
+                continue;
+            }
+
+            SinkRecordDescriptor sinkRecordDescriptor = buildRecordSinkDescriptor(record);
+
+            final TableId tableId = optionalTableId.get();
+
+            if (sinkRecordDescriptor.isTombstone()) {
```

Comment:
If you are taking about events processed by `ExtractNewRecordState` SMT, also this events produces tombstone with null value. So should be the same. Am I wrong?

Reply:
Yes and that's the problem. In that case tombstone == DELETE and you are dropping it. So deletes would never be processed by the task.

Reply:
Is true that a tombstone exists only if there is also the relative DELETE event? In that case the tombstone is skipped since the corresponding DELETE will be processed. Am I missing something?

Reply:
There is both DELETE and tomstone if full Envelpe Debezium messages are processed but if `ExtractNewRecordState` is use then only tombstone is provided so you will not create DELETE statement.

Reply:
So the case is when for `ExtractNewRecordState` with `delete.handling​.mode=none` the delete event will modified to have the `value = null`. But the method `isTombstone` will be true only if also the `valueSchema` is null.  The test `testShouldSkipTombstoneRecord` should cover the case both for Envelope and flat records. 

So seems that the Debezium `TOMBSTONE` has both `value` and `valueSchema` to `null`, instead the `ExtractNewRecordState` with `delete.handling​.mode=none`  will generate a record only with `value` null that by JDBC connector is treated as a flatten delete. 

Looking at the code seems we are missing the case when  `delete.handling​.mode` is `rewrite`. But I think we can manage it separately since is was already present before this PR. WDYT?

Reply:
This is very good finding regarding tombstone schema. Unfortunately now it creates a hidden dependency. Could you please add a comment to both this code and the SMT in question to make sure it is not lost.
WRT rewrite - I don't think we need to handle it for the time being.

Reply:
> Could you please add a comment to both this code and the SMT in question to make sure it is not lost. 

Yes, I can but this point my question is: should we align the behavior of 'ExtractNewRecordState' SMT to generate a correct 'TOMBSTONE' record top avoid future confusion? 

If yes, this will require a different task and just add the comments with this PR.

Reply:
I think it makes sense to have two types of tombstones exactly to handle this scenario. The tombstone generated by connector will be always ignored and by SMT will be processed. Otherwise we would need to intorudce a separate config option to control whether the tombstones should be ignored or not.

Reply:
I think that the user that want to use the `ExtractNewRecordState` with JDBC connector and process deletes, should set the new `delete.tombstone.handling.mode` to `rewrite` or `rewrite-with-tombstone` and then JDBC connector should skip the tombstone and, in case of flattened record, check the `__deleted` to understand if it's a delete or not. 

In any case I can proceed adding the comments and discuss this separately. Is it ok for you?

Reply:
Definitely!

Reply:
https://github.com/debezium/debezium/pull/4995